app/src/main/java/com/example/sharingang/UserProfileFragment.kt

```diff
@@ -32,22 +37,53 @@ class UserProfileFragment : Fragment() {
     private val itemsViewModel: ItemsViewModel by viewModels()
     private val args: UserProfileFragmentArgs by navArgs()
     private lateinit var binding: UserProfileFragmentBinding
+    private lateinit var authHelper: AuthHelper
 
     // This is the currently logged in user
     private var currentUserId: String? = null
     private lateinit var imageAccess: ImageAccess
 
+    private var currentUser: FirebaseUser? = null
+    private lateinit var userType: UserType
+
     // This is the user whose profile is shown (can be different from currentUserId)
     private var shownUserProfileId: String? = null
     private var loggedInUserEmail: String? = null
     private var imageUri: Uri? = null
 
+    private enum class UserType {
+        LOGGED_OUT_SELF,
+        LOGGED_OUT,
+        VISITOR,
+        SELF
+    }
+
     @Inject
     lateinit var currentUserProvider: CurrentUserProvider
 
     @Inject
     lateinit var userRepository: UserRepository
 
+    /*
+     * Launcher for the Login activity. Instead of using onActivityResult and startActivityForResult
+     * in a duo, we use this resultLauncher as it is a newer and more compact solution.
+     */
+    private val resultLauncher =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            if (result.resultCode == Activity.RESULT_OK) {
+                currentUserId = currentUserProvider.getCurrentUserId()
+                currentUser = currentUserProvider.getCurrentUser()
```

Comment:
Can this line return null somehow?

Reply:
Nope, if the login succeeds (meaning we have a RESULT_OK from the intent), then we are sure that our user and userId will not be null.